### PR TITLE
HPCC-24508 Include CPPUNIT in Docker images

### PR DESCRIPTION
**Pull request by richardkchapman**
At least in Debug images.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: dockerfiles/buildall.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ BUILD_TAG=$(git describe --exact-match --tags)  # The git tag for the images we
 BUILD_LABEL=${BUILD_TAG}                        # The docker hub label for all other components
 BUILD_USER=hpcc-systems                         # The github repo owner
 BUILD_TYPE=                                     # Set to Debug for a debug build, leave blank for default (RelWithDebInfo)
+INCLUDE_CPPUNIT=1
```
Comment (jakesmith) [inline dockerfiles/buildall.sh:28]: I think this is supposed to be USE_CPPUNIT?
"INCLUDE_CPPUNIT" isn't used anywhere afaics

File: dockerfiles/buildall.sh [hunk 2]
```diff
@@ -38,6 +39,7 @@ if [[ -n ${INPUT_BUILDTYPE} ]] ; then
 else
   BUILD_LABEL=${BUILD_TAG}
   BUILD_TYPE=RelWithDebInfo
+  USE_CPPUNIT=0
```
Comment (jakesmith) [inline dockerfiles/buildall.sh:42]: just to check - you're intentionally turning off, in the default case, when no INPUT_BUILDTYPE has been specified?
i.e. it's left on if the INPUT_BUILDTYPE is anything, including release?

**Comment by jakesmith** [comment]
@richardkchapman - a couple of questions.

File: dockerfiles/buildall.sh (hunk 1 above)
Comment (richardkchapman) [reply dockerfiles/buildall.sh:28]: Good catch. Have fixed

File: dockerfiles/buildall.sh (hunk 2 above)
Comment (richardkchapman) [reply dockerfiles/buildall.sh:42]: My goal was that the "standard" builds with no suffix on the build label did not include cppunit (though to be honest it's arguable whether they should and probably harmless for them to do so). Any "non-standard" build is for developer use and should include. I don't have strong feelings though - we could just set on all the time and simplify things significantly

Comment (jakesmith) [reply dockerfiles/buildall.sh:42]: @richardkchapman - minor image size increase with cppunit libs included? probably insignificant.

I think it's fine as it stands.

**Review by jakesmith — APPROVED** [review]

**Comment by richardkchapman** [comment]
@ghalliday

**Pull request merged**